plugin/trino-ignite/src/main/java/io/trino/plugin/ignite/IgniteJdbcClient.java

```diff
@@ -0,0 +1,655 @@
+
+        try (PreparedStatement preparedStatement = connection.prepareStatement(sql);
+                ResultSet resultSet = preparedStatement.executeQuery()) {
+            checkArgument(resultSet.next(), "Ignite table: '" + tableName + "' properties is NULL");
+            List<String> primaryKeys = extractColumnNamesFromOrderingScheme(resultSet.getString("PKS"));
+            checkArgument(!primaryKeys.isEmpty(), "Ignite table should has at least one primary key");
+            properties.put(PRIMARY_KEY_PROPERTY, primaryKeys);
+
+            for (String property : IgniteTableProperties.WITH_PROPERTIES) {
+                switch (property) {
+                    case AFFINITY_KEY_PROPERTY:
+                        List<String> affinityKeys = extractColumnNamesFromOrderingScheme(resultSet.getString("AFK"));
+                        if (!affinityKeys.isEmpty()) {
+                            String affinityKey = affinityKeys.get(0);
+                            checkArgument(ImmutableSet.copyOf(primaryKeys).contains(affinityKey), "Table affinity key should be one of the primary key");
+                            properties.put(property, affinityKey);
+                        }
+                        break;
+                    case BACKUPS_PROPERTY:
+                        int backups = resultSet.getInt(property.toUpperCase(ENGLISH));
+                        if (backups > 0) {
+                            properties.put(property, backups);
+                        }
+                        break;
+                    case TEMPLATE_PROPERTY:
+                        Optional.ofNullable(resultSet.getString(property.toUpperCase(ENGLISH)))
+                                .map(IgniteTableProperties.IgniteTemplateType::valueOf)
+                                .ifPresent(value -> properties.put(property, value));
+                        break;
+                    case WRITE_SYNCHRONIZATION_MODE_PROPERTY:
+                        Optional.ofNullable(resultSet.getString(property.toUpperCase(ENGLISH)))
+                                .map(IgniteTableProperties.IgniteWriteSyncMode::valueOf)
+                                .ifPresent(value -> properties.put(property, value));
+                        break;
+                    case CACHE_NAME_PROPERTY:
+                    case CACHE_GROUP_PROPERTY:
+                    case DATA_REGION_PROPERTY:
+                        Optional.ofNullable(resultSet.getString(property.toUpperCase(ENGLISH)))
+                                .ifPresent(value -> properties.put(property, value));
+                        break;
+                    default:
+                        throw new IllegalStateException("Unexpected value: " + property);
+                }
+            }
+        }
+        catch (SQLException e) {
+            throw new TrinoException(JDBC_ERROR, e);
+        }
+        return properties.build();
+    }
+
+    // extract result from format: "ID" ASC, "CITY_ID" ASC
```

Comment:
nit: What do you about the more generic `<column_name> <ASC|DESC>` instead?

Reply:
How about build a class to describe this, and we can do some check there also. what's your opinion

Reply:
Hi, about https://github.com/trinodb/trino/pull/8323#discussion_r702454753.
For the `query`, we can override the `prepareQuery` just as Druid connector does. But Ignite also support `ALTER` or `DROP`, which I do think will go through the `prepareQuery`. see `BaseJdbcClient#addColumn`, the `quoted` function still need to be override because Ignite do not recognize `IGNITE.SCHEMA.TABLE`.